mlserver/logging.py

```diff
@@ -35,13 +36,35 @@ def apply_logging_file(logging_settings: Union[str, Dict]):
         logger.warning("Unable to parse logging_settings.")
 
 
+class ModelLoggerFormatter(logging.Formatter):
+    """
+    A logging formatter that uses context variables to inject
+    the model name and version in the log message.
+    """
+
+    @staticmethod
+    def _fmt_model(name: str, version: str) -> str:
+        if not name:
+            return ""
+        model_fmt = f"{name}:{version}" if version else name
+        return f"[{model_fmt}]"
+
+    def format(self, record: logging.LogRecord) -> str:
+        model_name = model_name_var.get("")
+        model_version = model_version_var.get("")
+
+        record.model = self.__class__._fmt_model(model_name, model_version)
```

Comment:
Do you know if just `self._fmt_model` works in this case? Otherwise it may be worth just making the `_fmt_model` a regular instance method for simplicity.

Reply:
Yeah, that works. Thanks!